Source/GSStoryboardTransform.m

```diff
@@ -0,0 +1,828 @@
+                  forDocument: (NSXMLDocument *)document
+{
+  NSXMLNode *attr = [coel attributeForName: @"sceneMemberID"];
+  if ([[attr stringValue] isEqualToString: @"firstResponder"])
+    {
+      NSXMLNode *customClassAttr = [coel attributeForName: @"customClass"];
+      NSXMLNode *idAttr = [coel attributeForName: @"id"];
+      NSString *originalId = [idAttr stringValue];
+      
+      [idAttr setStringValue: @"-1"]; // set to first responder id
+      [customClassAttr setStringValue: @"FirstResponder"];
+      
+      // Actions
+      NSArray *cons = [document nodesForXPath: @"//action" error: NULL];
+      FOR_IN(NSXMLElement*, celem, cons) 
+        {
+          NSXMLNode *targetAttr = [celem attributeForName: @"target"];
+          NSString *val = [targetAttr stringValue];
+          if ([val isEqualToString: originalId])
+            {
+              [targetAttr setStringValue: @"-1"];
+            }
+        }
+      END_FOR_IN(cons);
+      
+      // Outlets
+      cons = [document nodesForXPath: @"//outlet" error: NULL];
+      FOR_IN(NSXMLElement*, celem, cons)
+        {
+          NSXMLNode *attr = [celem attributeForName: @"destination"];
+          NSString *val = [attr stringValue];
+          if ([val isEqualToString: originalId])
+            {
+              [attr setStringValue: @"-1"];
+            }
+        }
+      END_FOR_IN(cons);
+    }
+}
+
+- (void) processStoryboard: (NSXMLDocument *)xml
+{
+  NSArray *docNodes = [xml nodesForXPath: @"document" error: NULL];
+
+  if ([docNodes count] > 0)
+    {
+      NSXMLElement *docNode = [docNodes objectAtIndex: 0];
+      NSArray *array = [docNode nodesForXPath: @"//scene" error: NULL];
+      
+      // Set initial view controller...
+      ASSIGN(_initialViewControllerId, [[docNode attributeForName: @"initialViewController"] stringValue]);             
+      FOR_IN(NSXMLElement*, e, array) 
+        {
+          NSXMLElement *doc = [[NSXMLElement alloc] initWithName: @"document"];
+          NSArray *children = [e children];
+          NSXMLDocument *document = nil;
+          NSString *sceneId = [[e attributeForName: @"sceneID"] stringValue]; 
+          NSString *controllerId = nil;
+
+          // Move children...
+          FOR_IN(NSXMLElement*, child, children)
+            {
+              if ([[child name] isEqualToString: @"point"] == YES)
+                continue; // go on if it's a point element, we don't use that in the app...
+              
+              NSArray *subnodes = [child nodesForXPath: @"//application" error: NULL];
+              NSXMLNode *appNode = [subnodes objectAtIndex: 0];
+              [self processChild: child
+                         withDoc: doc
+                     withAppNode: appNode
+                         sceneId: sceneId];
+              
+              // fix other custom objects
+              document = [[NSXMLDocument alloc] initWithRootElement: doc]; // put it into the document, so we can use Xpath.
```

Comment:
The comment here is a bit surprising as you could use nodesForXPath: on an NSXMLNode already.

Reply:
Removed.